routes/apiRouter.js

```diff
@@ -19,208 +19,197 @@ routes.get('/', (req, res) => res.json({ message: 'Welcome to Imperial Bin\'s AP
+
+    if(!req.headers.authorization || !req.body.apiToken) return guestPaste();
+    const apiToken = req.headers.authorization || req.body.apiToken;
+    
+    Users.findOne({ apiToken }, (err, user) => {
+        if(err) return throwApiError(res, "Sorry! There was a internal server error, please contact a administrator!");
+        if(!user) return guestPaste();
+
+        const creator = user._id.toString();
+        const userSettings = {
+            longerUrls: req.body.longerUrls || false,
+            imageEmbed: req.body.imageEmbed || false,
+            expiration: req.body.expiration || 5,
+            instantDelete: req.body.instantDelete || false,
+            quality: !user.memberPlus ? 73 : 100 
+        }
+
+        var str = "";
+        // Short URLs are 8 characters long.
+        str += Math.random().toString(36).substr(2, 8);
+        // Long URLs are 15 characters long.
+        if(userSettings.longerUrls) str += Math.random().toString(36).substring(2, 7);
+        // Max duration.
+        if(userSettings.expiration > 31) userSettings.expiration = 31;
+
+        createPaste(str, userSettings.imageEmbed, userSettings.instantDelete, userSettings.expiration, creator, userSettings.quality);
+    });
+
     function createPaste(str, imageEmbed, instantDelete, expiration, creator, quality) {
+        if(!code) return throwApiError(res, "You need to post code! No code was submitted.");
         try {
-            if (code) {
-                db.link.insert({ URL: str, imageEmbed, instantDelete, creator, code, dateCreated: new Date().getTime(), deleteDate: new Date().setDate(new Date().getDate() + Number(expiration)), allowedEditor: [] }, (err, doc) => {
-                    res.json({
-                        success: true,
-                        documentId: str,
-                        rawLink: `https://www.imperialb.in/r/${str}`,
-                        formattedLink: `https://www.imperialb.in/p/${str}`,
-                        expiresIn: new Date(doc.deleteDate),
-                        instantDelete: instantDelete
-                    })
-                    if (quality && !instantDelete && imageEmbed) {
-                        screenshotDocument(str, quality);
-                    }
-                })
-            } else {
-                return throwApiError(res, "You need to post code! No code was submitted!");
-            }
-        } catch (err) {
-            return throwApiError(res, "An internal server error occurred, please contact an admin!");
-        }
+            db.link.insert({ URL: str, imageEmbed, instantDelete, creator, code, dateCreated: new Date().getTime(), deleteDate: new Date().setDate(new Date().getDate() + Number(expiration)), allowedEditor: [] }, (err, doc) => {
+                if(err) return throwApiError(res, "Sorry! There was a internal server error, please contact a administrator!");
+
```

Comment:
Write here could you add `await Users.updateOne({ _id: creator }, { $inc: { documentsMade: 1 } })` for a new feature.

Reply:
Implemented.